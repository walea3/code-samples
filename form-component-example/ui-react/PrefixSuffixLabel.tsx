import React, { ReactNode } from 'react';

import { getDataRef } from '@nel-ui/js-helpers';
import FormStyles from './FormStyles';

type HTMLDivAttr = React.HTMLAttributes<HTMLDivElement>;

export interface PrefixSuffixLabelProps extends HTMLDivAttr {
  id: string;
  children: ReactNode;
  'data-ref'?: string;
  isSuffix?: boolean;
  onClick: (event: React.MouseEvent<HTMLDivElement>) => void;
}

export function PrefixSuffixLabel({
  isSuffix = false,
  'data-ref': dataRef,
  ...props
}: PrefixSuffixLabelProps) {
  const suffixType = isSuffix ? 'Suffix' : 'Prefix';

  return (
    <FormStyles>
      {({ classes }) => (
        <div
          {...props}
          className={classes[`input${suffixType}`]}
          data-ref={getDataRef(dataRef, `input${suffixType}`)}
        />
      )}
    </FormStyles>
  );
}
