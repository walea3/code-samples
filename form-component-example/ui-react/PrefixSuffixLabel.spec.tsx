import { mountWithNEL, renderWithNEL } from '@nel-ui/tooling-react/test-utils';
import { ReactWrapper } from 'enzyme';
import React from 'react';
import { PrefixSuffixLabel, PrefixSuffixLabelProps } from './PrefixSuffixLabel';

export const findDataRef = (node: ReactWrapper, targetRef: string): ReactWrapper =>
  node.find(`[data-ref="${targetRef}"]`);

describe('<PrefixSuffixLabel />', () => {
  const requiredProps = {
    id: 'example',
    children: '',
    onClick: jest.fn(),
  };

  const mountComponent = (props?: Partial<PrefixSuffixLabelProps>) =>
    mountWithNEL(<PrefixSuffixLabel {...requiredProps} {...props} />);
  const renderComponent = (props?: Partial<PrefixSuffixLabelProps>) =>
    renderWithNEL(<PrefixSuffixLabel {...requiredProps} {...props} />);

  describe('@renders', () => {
    test.each`
      name                            | props
      ${'as default'}                 | ${{}}
      ${'with isSuffix'}              | ${{ isSuffix: true }}
      ${'with data-ref'}              | ${{ 'data-ref': 'uid' }}
      ${'with data-ref and isSuffix'} | ${{ 'data-ref': 'uid', isSuffix: true }}
      ${'with other props'}           | ${{ 'data-custom': 'uid' }}
    `(`$name`, ({ name, props }) => {
      expect(renderComponent({ ...props, children: name })).toMatchSnapshot();
    });
  });

  describe('@events', () => {
    test('should call onClick when clicked', () => {
      const { root, wrapper } = mountComponent();

      const el = findDataRef(wrapper, 'inputPrefix');
      el.simulate('click');

      expect(requiredProps.onClick).toHaveBeenCalledTimes(1);

      root.unmount();
    });
  });
});
