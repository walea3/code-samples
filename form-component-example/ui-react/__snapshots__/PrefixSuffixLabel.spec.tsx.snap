// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrefixSuffixLabel /> @renders as default 1`] = `
<div
  class="nel-inputPrefix nel-inputPrefixSuffix"
  data-ref="inputPrefix"
  id="example"
>
  as default
</div>
`;

exports[`<PrefixSuffixLabel /> @renders with data-ref 1`] = `
<div
  class="nel-inputPrefix nel-inputPrefixSuffix"
  data-ref="uid.inputPrefix"
  id="example"
>
  with data-ref
</div>
`;

exports[`<PrefixSuffixLabel /> @renders with data-ref and isSuffix 1`] = `
<div
  class="nel-inputSuffix nel-inputPrefixSuffix"
  data-ref="uid.inputSuffix"
  id="example"
>
  with data-ref and isSuffix
</div>
`;

exports[`<PrefixSuffixLabel /> @renders with isSuffix 1`] = `
<div
  class="nel-inputSuffix nel-inputPrefixSuffix"
  data-ref="inputSuffix"
  id="example"
>
  with isSuffix
</div>
`;

exports[`<PrefixSuffixLabel /> @renders with other props 1`] = `
<div
  class="nel-inputPrefix nel-inputPrefixSuffix"
  data-custom="uid"
  data-ref="inputPrefix"
  id="example"
>
  with other props
</div>
`;
